aten/src/ATen/BatchingRegistrations.cpp

```diff
@@ -0,0 +1,87 @@
+#include <ATen/BatchedTensorImpl.h>
+#include <torch/library.h>
+#include <ATen/BatchingUtils.h>
+#include <ATen/ATen.h>
+
+namespace at {
+
+// Note: [How to write batching rules]
+//
+// Whenever you do a vmap, on a vmap'ed tensor, the dimension that is being vmap'ed
+// gets recorded as a "batch dimension". Tensors that are not being vmap'ed over
+// behave as if they had a broadcasting batch dimension added to them.
+//
+// A batching rule function implements the logic of how to call an operator on
+// some inputs that have one or more batch dimensions.
+//
+// ==========================================
+// When and why should I add a batching rule?
+// ==========================================
+// When you are adding a new operator, you'll need to add a batching rule so
+// that vmap can work efficiently with said operator. If you do not, we'll attempt
+// to generate a slow fallback for the batching rule (TODO). In the future,
+// if the operator is composite (with respect to autograd), then we'll offer a
+// mechanism to register a fallthrough where we just run the composite op directly
+// on the BatchedTensor (TODO).
+//
+// ====================================
+// How to implement a new batching rule
+// ====================================
+// The signature of a batching rule should look like exactly like the C++ signature
+// of its operator.
+// `
+// Most batching rules will look like the following:
+// 1. Move all batch dims to the front of the input tensors
+// 2. If there are multiple tensors, align their batch dimensions by level
```

Comment:
I'd expand a bit on this sentence, since level wasn't introduced yet.

Reply:
Level is introduced in BatchedTensorImpl.h.https://github.com/pytorch/pytorch/blob/baf6ed0238df7a8f30655df5b3b265e77787e422/aten/src/ATen/BatchedTensorImpl.h#L19-L22

 But I can make that a note and refer back to it